### ISSUE TITLE
Disable Telemetry

### ISSUE THREAD

**Issue by JeremyWildsmith** [mono/mono#16875]
Can someone please suggest a way to disable any sort of telemetry at least? At best I would also like to disable crash reporting.

Is there any way to do this?

The main reason I want to do this is because the application spends a lot of time collecting / dumping and communicating telemetry data which is not suitable for my application.

Thanks.

**Comment by JeremyWildsmith** [comment]
I tried calling DisableMicrosoftTelemetry:

````
            var monoType = Type.GetType("Mono.Runtime", false);
            var m = monoType.GetMethod("DisableMicrosoftTelemetry", BindingFlags.NonPublic | BindingFlags.Static);
            var m_params = new object[] { };
            m.Invoke(null, m_params);
````

But I am getting an icall assertion:

````
Assertion: should not be reached at icall.c:5716
````

I am using the following install steps inside of my docker container that does this:
````
ENV MONO_VERSION 6.0.0.313
RUN apt-get update \
  && apt-get install -y --no-install-recommends gnupg dirmngr \
  && rm -rf /var/lib/apt/lists/* \
  && export GNUPGHOME="$(mktemp -d)" \
  && gpg --batch --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 3FA7E0328081BFF6A14DA29AA6A19B38D3D831EF \
  && gpg --batch --export --armor 3FA7E0328081BFF6A14DA29AA6A19B38D3D831EF > /etc/apt/trusted.gpg.d/mono.gpg.asc \
  && gpgconf --kill all \
  && rm -rf "$GNUPGHOME" \
  && apt-key list | grep Xamarin \
  && apt-get purge -y --auto-remove gnupg dirmngr

RUN echo "deb http://download.mono-project.com/repo/debian stable-stretch/snapshots/$MONO_VERSION main" > /etc/apt/sources.list.d/mono-official-stable.list \
  && apt-get update \
  && apt-get install -y mono-complete \
  && rm -rf /var/lib/apt/lists/* /tmp/*
````

**Comment by JeremyWildsmith** [comment]
I managed to disable telemetry for the specific signal causing my application to crash, but I don't think it is a good approach, but until I know of a better way I will need to do it this way.

````
            //Otherwise Mono will transit a lot of telemetry data which is time consuming
            Stdlib.SetSignalAction(Signum.SIGSEGV, SignalAction.Ignore);
````
Crashes significantly quicker, doesn't hang on trying to transmit telemetry data and bypasses crash reporting altogether.

**Comment by lambdageek** [comment]
I think this is about crash reporting, not telemetry.  (Calling `DisableMicrosoftTelemetry` is failing because the linux packages don't build with telemetry enabled).

Right now we only have a `configure` flag to disable crash reporting `configure --disable-crash-reporting`, but we should probably have an environment variable, too.

**Comment by JeremyWildsmith** [comment]
> I think this is about crash reporting, not telemetry. (Calling `DisableMicrosoftTelemetry` is failing because the linux packages don't build with telemetry enabled).
> 
> Right now we only have a `configure` flag to disable crash reporting `configure --disable-crash-reporting`, but we should probably have an environment variable, too.

By configure, do you mean on compile time? Or can I provide that configuration flag to the mono command?

**Comment by CoffeeFlux** [comment]
Yes, at compile time. `configure` is one of the scripts run in the compilation process.

**Comment by JeremyWildsmith** [comment]
> Yes, at compile time. configure is one of the scripts run in the compilation process.

Thanks for clarifying, I assumed so but I wanted to be sure.

In that case, this doesn't really work for me. We will need to hijack the SigSegv handler which is unfortunate because I don't feel like it is a good approach but it seems like the only solution for now.

Thanks!
